src/analyzer/protocol/ssl/SSL.cc

```diff
@@ -8,6 +9,73 @@
 #include "zeek/analyzer/protocol/ssl/ssl_pac.h"
 #include "zeek/analyzer/protocol/ssl/tls-handshake_pac.h"
 
+#include <arpa/inet.h>
+#include <openssl/evp.h>
+#include <openssl/kdf.h>
+
+#define MSB(a) ((a>>8)&0xff)
+#define LSB(a) (a&0xff)
+
+static void fmt_seq(uint32_t num, u_char* buf)
+	{
+	memset(buf, 0, 8);
+	uint32_t netnum = htonl(num);
+	memcpy(buf+4, &netnum, 4);
+	}
+
+static void print_hex(std::string name, u_char* data, int len)
+	{
+	int i = 0;
+	printf("%s (%d): ", name.c_str(), len);
+	if (len > 0)
+		printf("0x%02x", data[0]);
+
+	for (i = 1; i < len; i++)
+		{
+		printf(" 0x%02x", data[i]);
+		}
+	printf("\n");
+	}
+
+static bool tls1_prf(const u_char *secret, int secret_len, const char* label,
+		const u_char* rnd1, int rnd1_len, const u_char* rnd2, int rnd2_len, u_char* out, size_t out_len)
+	{
+	// alloc buffers
+	EVP_PKEY_CTX *pctx = EVP_PKEY_CTX_new_id(EVP_PKEY_TLS1_PRF, NULL);
+	size_t seed_len = strlen(label) + rnd1_len + rnd2_len;
+	u_char *seed = (u_char*) malloc(seed_len);
+	u_char *ptr = seed;
+
+	// seed = label + rnd1 + rnd2
+	memcpy(ptr, label, strlen(label));
```

Comment:
Just to check - is it guaranteed that the label does not contain 0-bytes? Also - can this potentially just be done with std::strings - all of this is mostly string-concatenations?